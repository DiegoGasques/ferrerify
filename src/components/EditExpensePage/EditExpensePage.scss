.EditExpensePage__header {
  background: darken($secondary, 20%);
  padding: $m-size $m-size;
  margin: $l-size 0;
  -webkit-box-shadow: $main-shadow;
  box-shadow: $main-shadow;

  h2 {
    color: $snow;
    font-size: $h3;
  }
}
