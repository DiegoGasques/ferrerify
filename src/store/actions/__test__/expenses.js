import { addExpense, removeExpense, editExpense } from "../expenses.actions.js";

it("Test for addExpense action creators correct output", () => {
  const newExpense = {
    description: "New expense",
    note: "This is a big expense",
    createdAt: 1000000,
    amount: 15000
  };
  const action = addExpense(newExpense);

  expect(action).toEqual({
    type: "ADD_EXPENSE",
    expense: {
      ...newExpense,
      id: expect.any(String)
    }
  });
});

it("Test for removeExpense action creators correct output", () => {
  const id = "123";
  const action = removeExpense(id);
  expect(action).toEqual({
    type: "REMOVE_EXPENSE",
    id
  });
});

it("Test for editExpense action creators correct output", () => {
  const id = "123";
  const updates = { note: "The updated note" };
  const action = editExpense(id, updates);
  expect(action).toEqual({
    type: "EDIT_EXPENSE",
    id,
    updates
  });
});
