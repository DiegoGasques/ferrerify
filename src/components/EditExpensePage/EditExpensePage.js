import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "../ExpenseForm/ExpenseForm";
import { editExpense } from "../../store/actions/expenses.actions";
import { getExpenseById } from "../../store/selectors/expenses.selector";

export const EditExpensePage = ({ expense, editExpense, history }) => {
  const onSubmit = expense => {
    editExpense(expense);
    history.push("/");
  };

  return (
    <div className="EditExpensePage">
      <ExpenseForm {...expense} onSubmit={onSubmit} />
    </div>
  );
};

const mapStateToProps = (state, props) => ({
  expense: getExpenseById(state, props)
});

const mapDispatchToProps = (dispatch, { match }) => ({
  editExpense: expense => dispatch(editExpense(match.params.id, expense))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
