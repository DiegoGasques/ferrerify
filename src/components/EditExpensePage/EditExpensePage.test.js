import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { EditExpensePage } from "./EditExpensePage";
import { expensesArr } from "../../fixtures/expenses";

describe("[COMPONENT] EditExpensePage", () => {
  it("Tests for correct render", () => {
    const wrapper = shallow(
      <EditExpensePage
        expense={expensesArr[0]}
        editExpense={() => {}}
        history={{ push: () => {} }}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Tests for correct onSubmit call", () => {
    const edits = {
      description: "New Description",
      note: "New Note",
      amount: "67368",
      createdAt: expensesArr[0].createdAt
    };
    const editExpenseSpy = jest.fn();
    const historyPushSpy = jest.fn();
    const wrapper = shallow(
      <EditExpensePage
        expense={expensesArr[0]}
        editExpense={editExpenseSpy}
        history={{ push: historyPushSpy }}
      />
    );

    wrapper.find("ExpenseForm").prop("onSubmit")(edits);
    expect(editExpenseSpy).toHaveBeenCalledWith(edits);
    expect(historyPushSpy).toHaveBeenCalledWith("/");
  });
});
