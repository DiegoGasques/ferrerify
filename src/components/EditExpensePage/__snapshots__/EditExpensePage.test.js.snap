// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] EditExpensePage Tests for correct render 1`] = `
<div
  className="EditExpensePage"
>
  <ExpenseForm
    amount={150000}
    createdAt={0}
    description="Rent"
    id="1"
    note="Monthly payment"
    onSubmit={[Function]}
  />
</div>
`;
